t/README.md

```diff
@@ -0,0 +1,117 @@
+# `t`
+
+This directory contains one of the two types of tests that the Git LFS project
```

Comment:
These two paragraphs are great and I think they should be in the Git LFS repository, but I think it would be better if we made a top-level `TESTING.md` file and put them there. I think it's a bit confusing to talk about the unit tests here, since this directory strictly deals with integration testing.

If we did make a top-level file, this more general testing overview info will be more visible and easily accessible, and we can link to this more granular, integration test-specific documentation.